### ISSUE TITLE
Clip+Draw on parent plus Clip Only on child has different behavior on opengl and vulkan

### ISSUE THREAD

**Issue by Gnumaru** [godotengine/godot#74186]
### Godot version

4.0.stable

### System information

windows 10, nvidia graphics

### Issue description

Setting "Clip+Draw" on a parent and "Clip Only" on it's direct child is an easy way of creating masks for "carving out" part of the parent image, the same way light masks worked on godot 3.

But unfortunately it only works on vulkan. On opengl the child image is drawn all white.

On vulkan

![Screenshot (292)](https://user-images.githubusercontent.com/5783414/222254859-dee43dd0-8fce-48f9-be36-7f22a89c93f7.png)

On opengl

![Screenshot (293)](https://user-images.githubusercontent.com/5783414/222254912-105cb1be-6e93-41fc-ab5f-e1fb9d87578f.png)

### Steps to reproduce

1. On an empty scene, create a sprite and set clip_children to Clip+Draw
2. Create a new sprite as a child and set clip_children to Clip Only and make it smaller than the parent
3. Set the renderer to mobile or forward+ and notice how the child sprite "carves out" the parent sprite, showing what's behind the parent
4. Then set the renderer to compatibility and notice how the child sprite is drawn all white (except any transparent parts, which show the parent behind)

### Minimal reproduction project

[ClipNDrawOnParentPlusClipOnlyOnChildWorksDifferentOnOpenGl.zip](https://github.com/godotengine/godot/files/10864945/ClipNDrawOnParentPlusClipOnlyOnChildWorksDifferentOnOpenGl.zip)

**Comment by insomniacUNDERSCORElemon** [comment]
I feel like it probably isn't intended, as recursive clip just broke it before and that was known.

I customized the internal clip shader a bit (seen [here](https://github.com/godotengine/godot/pull/67043#issuecomment-1272957622)) to get it working with blend modes:

![clipping_enhanced_w_selfmodulate](https://user-images.githubusercontent.com/4397725/195219710-e785e80b-9251-4fd1-b11a-be6e02ea5277.png)

but it wasn't added (maybe for a technical reason, but it just seems like it was glossed over).

**Comment by insomniacUNDERSCORElemon** [comment]
Sorry for the double comment, but I believe this is related: `clip only` has ghosting again with the mobile renderer in 4.0 and 4.0.1 RC 1:

![Screenshot_2023-03-15_06-30-52](https://user-images.githubusercontent.com/4397725/225428558-25d406c3-752d-4ba6-a92b-74aba84c8f47.png)

Irrelevant note: the stretched appearance is caused by the vertex being further out. Otherwise it'd just look like a tiny extra view.

**Comment by gu-fan** [comment]
yes, on mobile  renderer, the clip behavior is wrong. on forward+ it works right.
